src/core/shiftManagement.ts

```diff
@@ -55,3 +62,47 @@ export function applyToShift(
     },
   });
 }
+
+export function rateWorker(
+  workerUuid: Worker["uuid"],
+  shiftUuid: Shift["uuid"],
+  rating: number,
+): Promise<ShiftAssignment> {
+  return prisma.shiftAssignment.update({
+    where: {
+      shiftUuid_workerUuid: {
+        shiftUuid,
+        workerUuid,
+      },
+    },
+    data: { rating },
+  });
+}
+
+export async function blockWorker(
+  workerUuid: Worker["uuid"],
+  shiftUuid: Shift["uuid"],
+  blockReason: string,
+): Promise<GenericSuccessResponse> {
+  const shift = await prisma.shift.findUniqueOrThrow({
+    where: {
+      uuid: shiftUuid,
+    },
+  });
+  const facility = await prisma.healthCareFacility.findUniqueOrThrow({
+    where: {
+      uuid: shift.facilityUuid,
+    },
+  });
+  await prisma.blockedWorker.create({
+    data: {
+      facilityUuid: facility.uuid,
+      shiftUuid,
+      workerUuid,
+      blockReason,
+      createdAt: new Date(),
+    },
+  });
+
+  return { status: "ok" };
+}
```

Comment:
The current implementation of `blockWorker` function can be simplified and optimized. Here are my observations and suggestions:

🎯 Main Concern:
- The function uses `shiftUuid` to indirectly fetch the facility, but blocking is a facility-level operation, not a shift-level one.

💡 Suggested Changes:
1. Modify the function signature to directly accept `facilityUuid` instead of `shiftUuid`:
```typescript
blockWorker(
  workerUuid: Worker["uuid"],
  facilityUuid: HealthCareFacility["uuid"],
  blockReason: string
)
```
- This should remove the unnecessary database queries for shift and facility lookup.

- [ ] 🚨 **Critical Issue**
The current implementation of `applyShift` doesn't implement the business requirement that blocked workers should not be able to apply to facility shifts. This could lead to blocked workers successfully applying to shifts, violating business rules.
This change is necessary to enforce the new blocking functionality requirement: _Workers who have been blocked from applying to a healthcare facility's shifts should not be able to apply to shifts from the facility that blocked them_
- [ ] Consider using prisma transactions to have better consistency in high-concurrency scenarios.

Reply:
@bikash119 the requirement clearly mentions to include shiftUuid in the request body. in my view, we shouldn't be going against requirements

Reply:
the highlight  point should be adding error handlings to the DB actions. no where in the code try/catch blocks used and proper errors not returned to the client